PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = ['onboarding.onboarding.step']
-    _inherit = ['res.company']
-    _inherit = ['res.country']
-    _inherit = ['res.partner']

Total Changes: 14

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/payment/models/ir_http.py
+++ b/addons/payment/models/ir_http.py
@@ -4,7 +4,7 @@ from odoo import models
 
 
 class IrHttp(models.AbstractModel):
-    _inherit = ['ir.http']
+    _inherit = 'ir.http'
 
     @classmethod
     def _get_translation_frontend_modules_name(cls):

--- a/addons/payment/models/onboarding_step.py
+++ b/addons/payment/models/onboarding_step.py
@@ -4,7 +4,7 @@ from odoo import api, models
 
 
 class OnboardingOnboardingStep(models.Model):
-    _inherit = ['onboarding.onboarding.step']
+    _inherit = 'onboarding.onboarding.step'
 
     @api.model
     def action_validate_step_payment_provider(self):

--- a/addons/payment/models/payment_method.py
+++ b/addons/payment/models/payment_method.py
@@ -10,6 +10,7 @@ from odoo.addons.payment.const import REPORT_REASONS_MAPPING
 
 
 class PaymentMethod(models.Model):
+    _name = 'payment.method'
     _description = "Payment Method"
     _order = 'active desc, sequence, name'
 

--- a/addons/payment/models/payment_provider.py
+++ b/addons/payment/models/payment_provider.py
@@ -13,6 +13,7 @@ _logger = logging.getLogger(__name__)
 
 
 class PaymentProvider(models.Model):
+    _name = 'payment.provider'
     _description = 'Payment Provider'
     _order = 'module_state, state desc, sequence, name'
     _check_company_auto = True

--- a/addons/payment/models/payment_token.py
+++ b/addons/payment/models/payment_token.py
@@ -5,6 +5,7 @@ from odoo.exceptions import UserError, ValidationError
 
 
 class PaymentToken(models.Model):
+    _name = 'payment.token'
     _order = 'partner_id, id desc'
     _description = 'Payment Token'
     _check_company_auto = True

--- a/addons/payment/models/payment_transaction.py
+++ b/addons/payment/models/payment_transaction.py
@@ -21,6 +21,7 @@ _logger = logging.getLogger(__name__)
 
 
 class PaymentTransaction(models.Model):
+    _name = 'payment.transaction'
     _description = 'Payment Transaction'
     _order = 'id desc'
     _rec_name = 'reference'

--- a/addons/payment/models/res_company.py
+++ b/addons/payment/models/res_company.py
@@ -4,7 +4,7 @@ from odoo import api, fields, models
 
 
 class ResCompany(models.Model):
-    _inherit = ['res.company']
+    _inherit = 'res.company'
 
     payment_onboarding_payment_method = fields.Selection(
         string="Selected onboarding payment method",

--- a/addons/payment/models/res_country.py
+++ b/addons/payment/models/res_country.py
@@ -6,7 +6,7 @@ import odoo.addons.payment_stripe as stripe  # prevent circular import error wit
 
 
 class ResCountry(models.Model):
-    _inherit = ['res.country']
+    _inherit = 'res.country'
 
     is_stripe_supported_country = fields.Boolean(compute='_compute_is_stripe_supported_country')
 

--- a/addons/payment/models/res_partner.py
+++ b/addons/payment/models/res_partner.py
@@ -4,7 +4,7 @@ from odoo import api, fields, models
 
 
 class ResPartner(models.Model):
-    _inherit = ['res.partner']
+    _inherit = 'res.partner'
 
     payment_token_ids = fields.One2many(
         string="Payment Tokens", comodel_name='payment.token', inverse_name='partner_id')
